website/project/model.py

```diff
@@ -3479,6 +3419,204 @@ def require_approval(self, user, notify_initiator_on_complete=False):
+                action=NodeLog.PRIMARY_INSTITUTION_CHANGED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    },
+                    'previous_institution': {
+                        'id': previous._id if previous else None,
+                        'name': previous.name if previous else 'None'
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def remove_primary_institution(self, user, log=True):
+        inst = self.primary_institution
+        if not inst:
+            return False
+        self.primary_institution = None
+        if inst in self.affiliated_institutions:
+            self.affiliated_institutions.remove(inst)
+        inst = Institution(inst)
+        if log:
+            self.add_log(
+                action=NodeLog.PRIMARY_INSTITUTION_REMOVED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def institution_url(self):
+        return self.absolute_api_v2_url + 'institution/'
+
+    def institution_relationship_url(self):
+        return self.absolute_api_v2_url + 'relationships/institution/'
+
+class SpecialList(list):
+    def __init__(self, node, private_target, init=None):
+        super(SpecialList, self).__init__(init or [])
+        self.node = node
+        self.target = private_target
+
+    def append(self, to_append):
+        junk = getattr(self.node, self.target)
+        junk.append(to_append.node)
+        setattr(self.node, self.target, junk)
+
+class InstitutionQuerySet(MongoQuerySet):
+    def __init__(self, queryset):
+        super(InstitutionQuerySet, self).__init__(queryset.schema, queryset.data)
+
+    def __iter__(self):
+        for each in super(InstitutionQuerySet, self).__iter__():
+            yield Institution(each)
+
+    def _do_getitem(self, index):
+        item = super(InstitutionQuerySet, self)._do_getitem(index)
+        if isinstance(item, MongoQuerySet):
+            return self.__class__(item)
+        return Institution(item)
+
+class Institution():
+
+    institution_node_translator = {
+        '_id': 'institution_id',
+        'auth_url': 'institution_auth_url',
+        'domain': 'institution_domain',
+        'name': 'title',
+        'logo_name': 'institution_logo_name'
+    }
+
+    def __init__(self, node):
+        if node is None:
+            return
+        self.node = node
+        for key, value in self.institution_node_translator.iteritems():
+            setattr(self, key, getattr(node, value))
+
+    def __getattr__(self, item):
+        return getattr(self.node, item)
```

Comment:
If ```self``` is an ```Institution``` then wouldn't ```getattr(self, '_id')``` incorrectly return ```self.node._id``` instead of ```self.node.institution_id```?